src/clientiface.cpp

```diff
@@ -699,7 +699,11 @@ void ClientInterface::UpdatePlayerList()
 		for (session_t i : clients) {
 			RemotePlayer *player = m_env->getPlayer(i);
 
-			if (player == NULL)
+			if (!player)
+				continue;
+
+			PlayerSAO *sao = player->getPlayerSAO();
+			if (sao && sao->isHidden())
```

Comment:
sorry but SAO has nothing to do with Player list. it's not responsibility of an AO property to update player list

Reply:
Do you have a better idea?